blockchain/chain/src/store/chain_store.rs

```diff
@@ -539,26 +543,39 @@ where
 
     /// Subscribes to head changes. This function will send the current tipset through a channel,
     /// start a task that listens to each head change event and forwards into the channel,
-    /// then returns the receiver of this channel from the function.
+    /// then returns an id corresponding to the receiver of this channel from the function.
     /// This function is not blocking on events, and does not stall publishing events as it will
     /// skip over lagged events.
-    pub async fn sub_head_changes(&self) -> Receiver<HeadChange> {
-        let (tx, rx) = bounded(16);
+    pub async fn sub_head_changes(&self) -> i64 {
+        let (tx, rx) = channel::unbounded();
         let mut subscriber = self.publisher.subscribe();
 
+        let sub_id = self.subscriptions.read().await.len() as i64;
+
+        self.subscriptions
+            .write()
+            .await
+            .insert(sub_id.clone(), Some(rx));
```

Comment:
@cryptoquick  I think we can remove `.clone()` from `sub_id` as `i64` is `Copy`

Reply:
@creativcoder Good point! That was leftovers from something else. It's always a good idea to minimize clones. :+1: